typescript/web/src/connectors/labeling-state/index.ts

```diff
@@ -55,6 +71,39 @@ export const useLabelingStore = create<LabelingState>(
     zoomFactor: 0.5,
     canZoomIn: true,
     canZoomOut: false,
+    iogSpinnerPosition: null,
+    iogProcessingLabels: new Set(),
+    iogSpinnerPositions: {},
+    registerIogJob: (timestamp, idLabel, iogSpinnerPosition) => {
+      if (!iogSpinnerPosition || !idLabel) return;
+      const { iogSpinnerPositions, iogProcessingLabels } = get();
+      iogProcessingLabels.add(idLabel);
+      const newIogSpinnerPositions = {
+        ...iogSpinnerPositions,
+        [timestamp]: iogSpinnerPosition,
+      };
+      // @ts-ignore See https://github.com/Diablow/zustand-store-addons/issues/2
+      set({
+        iogProcessingLabels,
+        iogSpinnerPositions: newIogSpinnerPositions,
+      });
+    },
+    unregisterIogJob: (timestamp, idLabel) => {
+      const { iogSpinnerPositions, iogProcessingLabels } = get();
+      if (idLabel) iogProcessingLabels.delete(idLabel);
+      const newIogSpinnerPositions = Object.fromEntries(
+        Object.keys(iogSpinnerPositions)
+          .filter(
+            (timestampCurrent) => parseInt(timestampCurrent, 10) !== timestamp
+          )
+          .map((key) => [key, iogSpinnerPositions[parseInt(key, 10)]])
+      );
+      // @ts-ignore See https://github.com/Diablow/zustand-store-addons/issues/2
+      return set({
+        iogSpinnerPositions: newIogSpinnerPositions,
+        iogProcessingLabels,
+      });
+    },
```

Comment:
Those functions seemed to introduce randomness somehow, I don't know if they're legitimate?